samples/nrf_desktop/src/modules/ble_adv.c

```diff
@@ -228,6 +231,7 @@ static void sp_grace_period_fn(struct k_work *work)
 	if (err) {
 		module_set_state(MODULE_STATE_ERROR);
 	} else {
+		state = STATE_OFF;
```

Comment:
Here we shouldn't go to STATE_OFF but to IDLE. Remember that grace period is scheduled from `ble_adv_stop()` which is called e.g. when we get PEER_STATE_CONNECTED.

Reply:
My understanding is that when we stop advertising, we go to IDLE (so advertising should be IDLE when BLE is connected) and that we go to OFF state only when power down event comes.

Reply:
discussed offline - grace_period timeout is scheduled from `remove_vendor_section` which is called on power down. When we reach line 234 it means that nobody connected to us during grace period and so we should continue with a system off.